src-tauri/src/helpers/check_admin_creds.rs

```diff
@@ -0,0 +1,225 @@
+    if unauthorized_endpoints.len() == 0 {
+        return Err("No unauthorized endpoints found".to_string());
+    }
+
+    // find a random
+    let first_endpoint = format!("{}{}", address, unauthorized_endpoints[0]);
+
+    if let Some(creds) = brute_force_credentials(first_endpoint.as_str(), client.clone()).await {
+        return Ok(Some(creds));
+    }
+    return Ok(None);
+}
+
+// Given a host and a port,
+// return a valid and appropriate http url
+fn create_address_url(host: &str, port: i32) -> String {
+    if port == 443 {
+        return format!("https://{}:{}", host, port);
+    } else {
+        return format!("http://{}:{}", host, port);
+    }
+}
+
+fn got_redirect_response(text: &str) -> bool {
+    // Currently, to check whether a client-side redirect was returned
+    // we simply check whether only a script was returned
+    return text.starts_with("<script>") && text.ends_with("</script>");
+}
+
+async fn get_redirected_response(
+    text: &str,
+    client: Client,
+    address: &str,
+) -> Result<Response, String> {
+    let url = capture_url_from_redirect_script(text);
+
+    match url {
+        None => {
+            println!("No match found for {}", text);
+            return Err("No redirect found".to_string());
+        }
+        Some(t) => {
+            // redirect url found, lets go there
+            let new_address = format!("{}{}", address, t);
+            let response = client
+                .get(&new_address)
+                .send()
+                .await
+                .map_err(|e| e.to_string());
+            return response;
+        }
+    }
+}
+
+fn capture_url_from_redirect_script(redirect_script: &str) -> Option<String> {
+    // regex which finds the url content wrapped in single quotes
+    let rx = Regex::new(r"url\s*=\s*'([^']+)'").unwrap();
+    return rx
+        .captures(redirect_script)
+        .and_then(|caps| caps.get(1).map(|match_| match_.as_str().to_string()));
+}
+
+// Finds and returns a vector of the contents of script tags like these:
+//  <script defer="defer" src="/main.1f13cbe8ee4a4f1a0848.js"></script>
+// a returned string should follow this format:
+// 'defer="defer" src="/main.1f13cbe8ee4a4f1a0848.js"'
+fn find_script_tags(http: &str) -> Vec<String> {
+    let rx = Regex::new(r"<script (.*?)></script>").unwrap();
+    let matches = rx
+        .captures_iter(http)
+        .filter_map(|cap| cap.get(1).map(|m| m.as_str().to_string()))
+        .collect();
+
+    return matches;
+}
+
+fn find_script_src(script: &str) -> Option<String> {
+    let rx = Regex::new("src\\s*=\\s*\"([^']+)\"").unwrap();
+    return rx
+        .captures(script)
+        .and_then(|caps| caps.get(1).map(|m| m.as_str().to_string()));
+}
+
+fn find_endpoints_from_code(code: &str) -> Vec<String> {
+    // find all possible endpoints within the javascript code
+    let regex_pattern = r#""(/[a-zA-Z0-9\/]+)""#;
+    let rx = Regex::new(regex_pattern).unwrap();
```

Comment:
Code might panic if regex is not initilized. This is unlikely but maybe better to handle with a result type which returns an error. Then this could look something like this: 

-> Result<Vec<String>, Err> 

let rx = Regex::new(regex_pattern)?;

Reply:
Also applies to the find_script_src